### ISSUE TITLE
Redirect issue when using a ChooserBlock

### ISSUE THREAD

**Issue by Nigel2392** [wagtail/wagtail-localize#759]
There is an issue when you select an item in synchronized pages.
When you select an item using a ChooserBlock, normally a JSON response is returned, and the object is updated in the frontend with javascript. This does not happen.

Instead, when you click an item of a `ChooserBlock` in a synchronized page, you will be forwarded to the JSON response page. This means, you will leave the regular admin area; and only see an ugly JSON response. 

Example:

https://github.com/wagtail/wagtail-localize/assets/91429854/de4b3f9c-5186-4432-bfc5-b21334fc8e16

**Comment by zerolab** [comment]
Thank you for the report @Nigel2392. One more reason to get to #715 ASAP
Will see if I can find a fix for this over the weekend

**Comment by dassudip2001** [comment]
@zerolab I think redirecting to another page shows a JSON response instead of showing a model and binding the JSON response as a table-like format

**Comment by dassudip2001** [comment]
if js error then can I fix this issue or is the model not present can I implement @zerolab

**Comment by zerolab** [comment]
@dassudip2001 I'm afraid you'd have to do the legwork for this as I don't know where the issue is (at least yet)

The test app has a good number of StreamField blocks using the chooser (e.g. https://github.com/wagtail/wagtail-localize/blob/main/wagtail_localize/test/models.py#L200-L209)

In theory you could even run the test app in a virtualenvironment to get a demo instance

**Comment by Nigel2392** [comment]
Some more info:
1. Logo's model is non-translatable.
2. Branding model is translatable.
3. When inspecting (network and console) and using the "Persist log" option - no errors seem to occur.

This gives me the impression that the JS callback isn't properly registered - possibly this happened when Wagtail's chooserblock was made more generic. 

The chain of events:
1. We make a request to 
`http://127.0.0.1:8080/admin/snippets/choose/<app_name>/<model_name>/`, this retrieves the modal html? No errors occur at this step.
```json
{"html": "\n\n\n<header class=\"w-header w-header--merged\">\n    <div class=\"row\">\n        <div class=\"left\">\n            <div class=\"col\">\n                \n                    <h1 class=\"w-header__title\" id=\"header-title\">\n                        \n                            <svg class=\"icon icon-image w-header__glyph\" aria-hidden=\"true\"><use href=\"#icon-image\"></use></svg>\n                        \n                        Kies <span class=\"w-header__subtitle\">Logo&#x27;s</span>\n                    </h1>\n                \n                \n            </div>\n            \n        </div>\n        <div class=\"right\">\n            \n                \n            \n            \n        </div>\n    </div>\n    \n</header>\n\n\n\n\n\n<div class=\"w-tabs\" >\n    \n        \n        <div class=\"w-tabs__wrapper\"></div>\n    \n\n    <div class=\"tab-content\">\n        <section\n            id=\"tab-search\"\n            class=\"w-tabs__panel\"\n            role=\"tabpanel\"\n            aria-labelledby=\"tab-label-search\"\n        >\n\n            \n                \n            \n\n            \n                <div id=\"search-results\" class=\"listing\">\n                    \n\n\n    \n        \n    \n\n    \n        \n\n<table class=\"listing\">\n    \n    <thead>\n        <tr >\n            \n                \n<th >\n    \n        Titel\n    \n    \n</th>\n\n            \n        </tr>\n    </thead>\n    <tbody>\n        \n            <tr>\n                \n                    \n\n<td class=\"title\">\n    <div class=\"title-wrapper\">\n        \n            \n                <a  data-chooser-modal-choice href=\"/admin/snippets/choose/blockapp/logos/chosen/1/\">Branding</a>\n            \n        \n    </div>\n    \n</td>\n\n                \n            </tr>\n        \n    </tbody>\n</table>\n\n    \n    \n\n\n\n\n\n<nav class=\"pagination\" aria-label=\"Pagineren\">\n    <p>Pagina 1 van 1.</p>\n    <ul>\n        <li class=\"prev\">\n            \n        </li>\n        <li class=\"next\">\n            \n        </li>\n    </ul>\n</nav>\n\n\n\n                </div>\n            \n        </section>\n\n        \n    </div>\n</div>\n", "step": "choose"}
```

2. When chosen - the page immediately goes to the new JSON view (`http://127.0.0.1:8080/admin/snippets/choose/<app_name>/<model_name>/chosen/<model_id>/`).  
There is no redirect.  There are no errors.
This means there is likely no callback being registered when the modal HTML gets appended to the document.
It happens for both translatable and non translatable models, likely since they use the same `Chooser` class underneath.

Regular chooser chain of events are now over.

1. For images it goes to `http://127.0.0.1:8080/admin/images/chooser/`.
This retrieves the modal HTML - there might be a different mechanism of adding eventlisteners to image-chooser modals.

2. Then when selecting an image - there is an eventlistener registered. It uses ajax to fetch to `http://127.0.0.1:8080/admin/images/chooser/chosen/<image_id>/`.  
This retrieves the following JSON object: 
```json
{
    "step": "chosen",
    "result": {
        "id": "2843",
        "title": "csm_SR_0012_6b12b53339",
        "edit_url": "/admin/images/2843/",
        "preview": {
            "url": "/media/images/csm_SR_0012_6b12b53339.max-165x165.jpg",
            "width": 165,
            "height": 109
        }
    }
}
```
3. Then from javascript - a PUT request gets made to update the synchronized page:  
(`http://127.0.0.1:8080/admin/localize/translate/<model_id>/overrides/<segment_id>/edit/`).  
The eventlisteners seem to be properly registered to the modal.
```
{
    "segment_id": 578,
    "data": "2843",
    "error": null
}
```
Image-chooser chain of events is now over.

**Comment by PaoloC68** [comment]
I have the same issue and I just wonder if there is any solution, even temporary? 
![image](https://github.com/wagtail/wagtail-localize/assets/1483774/4278b639-6045-49c7-8ee4-0c03ac06dfa4)

**Comment by zerolab** [comment]
I've been looking into this.
tl;dr at some point pre 5.2 the snippet chooser was updated to pretty much be as generic as it comes.

We do https://github.com/wagtail/wagtail-localize/blob/f0056012d36c67c2ba415baff0fdc3e87378541f/wagtail_localize/static_src/editor/components/TranslationEditor/segments.tsx#L736-L749, but `window.SNIPPET_CHOOSER_MODAL_ONLOAD_HANDLERS` is no longer there, so none of the default chooser handlers are run (init search, ajaxifying links etc)

Looking at a workaround

**Comment by zerolab** [comment]
Fixed and released v1.9 with full Wagtail 6 support
